frontstage/views/account/accept_transfer_survey.py

```diff
@@ -0,0 +1,119 @@
+import logging
+
+from flask import render_template, abort, request, url_for
+from structlog import wrap_logger
+from werkzeug.utils import redirect
+
+from frontstage import app
+from frontstage.common.authorisation import jwt_authorization
+from frontstage.controllers import party_controller, survey_controller
+from frontstage.controllers.party_controller import get_business_by_id
+from frontstage.exceptions.exceptions import ApiError
+from frontstage.views.account import account_bp
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+@account_bp.route('/transfer-surveys/accept-transfer-surveys/<token>', methods=['GET'])
+def get_transfer_survey_summary(token):
+    """
+    Endpoint to verify transfer token and retrieve the summary page
+    :param token: transfer survey token
+    :type token: str
+    """
+    logger.info('Getting transfer survey summary', token=token)
+    try:
+        response = party_controller.verify_pending_survey_token(token)
+        pending_transfer_surveys = response.json()
+        transfer_dict = {}
+        distinct_businesses = set()
+        batch_number = pending_transfer_surveys[0]['batch_no']
+        originator_party = party_controller.get_respondent_party_by_id(pending_transfer_surveys[0]['shared_by'])
+        transferred_by = originator_party['emailAddress']
+        for pending_transfer_survey in pending_transfer_surveys:
+            distinct_businesses.add(pending_transfer_survey['business_id'])
```

Comment:
I'm aware it shouldn't be possible for us to have the same business across multiple `pending_transfer_surveys`, but it might be best to check if `distinct_businesses` already contains this business ID so that it's definitely a list of _distinct_ businesses

Reply:
🎨

Reply:
At this point of time.. it will be definitely a distinct business.